src/db.c

```diff
@@ -5234,7 +5243,7 @@ db_queue_add_by_query(struct query_params *qp, char reshuffle, uint32_t item_id,
     goto end_transaction;
 
   // Reshuffle after adding new items
-  if (reshuffle)
+  if (position == -1 && reshuffle)
```

Comment:
It would be better to have the same condition as in line 5193. We modify the `queue_count` value, so cannot use it here. Maybe best to introduce a variable that stores the condition result. And extend the existing comment to explain a bit why we do not always reshuffle, even when the parameter shuffle is true.

```
  bool append_to_queue;

  ...

  append_to_queue = (position < 0 || position > queue_count);

  if (append_to_queue)
    {
      pos = queue_count;
      ...
    }
  else
    {
      pos = position;
      ...
    }

  // Reshuffle after adding new items, if no queue position was specified
  if (append_to_queue && reshuffle)
    {
      ret = queue_reshuffle(item_id, queue_version);
    }
```

Reply:
thanks - updated to match review